cla_backend/libs/eligibility_calculator/calculator.py

```diff
@@ -203,32 +208,29 @@ def is_disposable_income_complete(case_data):
 
     @staticmethod
     def is_property_complete(case_data):
-        if not hasattr(case_data.facts, "has_partner"):
+        if "has_partner" not in case_data.facts.__dict__:
             # If they have a partner then that may increase assets that they need to delare, so
             # this section is not complete until we clear up if there is a partner
             return False
 
         if len(case_data.property_data) == 0:
             return True
-        for key, value in case_data.property_data[0].iteritems():
+        for key, value in case_data.property_data[0].items():
             if value is not None:
                 return True
         return False
 
     @staticmethod
-    def is_savings_complete(case_data):
-        if not hasattr(case_data.facts, "has_partner"):
+    def _is_savings_complete(case_data):
+        if "has_partner" not in case_data.facts.__dict__:
             # If they have a partner then that may increase assets that they need to delare, so
             # this section is not complete until we clear up if there is a partner
             return False
 
-        try:
-            savings = case_data.you.savings
-            for key in savings.PROPERTY_META:
-                if not isinstance(getattr(savings, key), (types.IntType, types.LongType)):
-                    return False
-        except exceptions.PropertyExpectedException:
-            return False
+        savings = case_data.you.savings
+        for key in savings.PROPERTY_META:
+            if not isinstance(getattr(savings, key), (types.IntType, types.LongType)):
+                return False
```

Comment:
Testing revealed an issue with this method. When savings has missing keys, we need it to return False. Previously getattr would raise the PropertyExpectedException, however the new behaviour of getattr is to return default values. I think we need:
```
if key not in savings.__dict__:
    return False
if not isinstance(getattr(savings, key), (types.IntType, types.LongType)):
    # usually: getattr(savings, key) == None
    return False
```